test/network_policy_test.go

```diff
@@ -62,12 +64,271 @@ var _ = Describe(
 				)
 			},
 		)
+
```

Comment:
Few more tests:

1. HostNetworking related validation
2. Network Namespace different from CR
3. Invalid network name given in both (pod annotations and network policy) should fail
4. Max, Min networkName len 1
5. Fabric networkNames update should fail
6. Recovery test: pass wrong networkNames, pod should stuck, then update CR with right networkNames, pod should succeed.

Reply:
4.  I don't think these test-cases are required as these limits are handled at CRD schema level
5. Not added because can't test with fabric interfaces at the moment as it fails in cluster create flow. Reason is it tries to bind the random IPs provided in the annotations

Reply:
4. I think tests should not be dependent on how the validations are happening internally. It should mainly care about the expectation made to the user. If validations are removed or changed in the future from CRD then tests should be able to catch that.